EIPS/eip-7620.md

```diff
@@ -243,18 +250,40 @@ function txcreate(a, b, c, d, e) -> f {
     
 ## Rationale
 
+### Data section appending
+
 The data section is appended to during contract creation and also its size needs to be updated in the header. Alternative designs were considered, where:
 
 - additional section kinds for the data were introduced
 - additional fields describing a subcontainer were introduced
 - data section would be written over as opposed to being appended to, requiring it to be filled with 0 bytes prior to deployment
     
 All of these alternatives either complicated the otherwise simple data structures or took away useful features (like the dynamically sized portion of the data section).
+
+### `TXCREATE`
+
+An alternative design was discussed, where instead of `TXCREATE`, `InitcodeTransaction` and the Creator Contract, the legacy-like creation transactions (with empty `to`) continue to be available to send EOF initcontainers in transactions and perform deployment of EOF contracts.
+
+This makes it impossible for smart contract wallets to deploy arbitrary contracts (only EOAs can), which is a desired feature of current legacy creation rules. A workaround where those arbitrary contracts are first "uploaded" to a factory contract, and then deployed using `EOFCREATE`, is not satisfactory.
+
+### `TXCREATE` failure modes
     
 `TXCREATE` has two "light" failure modes in case the initcontainer is not present and in case the EOF validation is unsuccessful. An alternative design where both cases led to a "hard" failure (consuming the entire gas available) was considered. We decided to have the more granular and forgiving failure modes in order to align the gas costs incurred to the actual work the EVM performs.
 
+### Creator Contract
+
 EOF contract creation requires the Creator Contract be introduced via a state change, because neither legacy contracts nor create transactions can deploy EOF code. The alternative approach which was to continue using legacy creation would still rely on fetching the *initcode* from memory and not satisfy the requirement of code non-observability.
 
+### EOF validation checking initcode-mode
+
+Extra EOF validation rules were considered:
+
+- Each EOF subcontainer must either be referenced by an `EOFCREATE` or a `RETURNCONTRACT` instruction, but never both
+- Each EOF subcontainer may either contain `RETURNCONTRACT` or `RETURN` / `STOP` instructions, but never a mix of these two kinds
+- A subcontainer referenced by an `EOFCREATE` must never contain a `RETURN` / `STOP` instruction, whereas a subcontainer referenced by a `RETURNCONTRACT` must never contain a `RETURNCONTRACT`
+
+Given no obvious merit of enforcing these rules, it is preferred to avoid the extra logic and complexity (esp. the last rule causes EOF validation of subcontainers to not be self-contained).
```

Comment:
Agree there is merit
```suggestion
These rules were not adopted to avoid the extra logic and complexity (esp. the last rule causes EOF validation of subcontainers to not be self-contained).
```